pdns/mtasker.cc

```diff
@@ -325,7 +325,14 @@ template<class Key, class Val, class Cmp>bool MTasker<Key,Val,Cmp>::schedule(con
     d_threads[d_tid].dt.start();
 #endif
     notifyStackSwitch(d_threads[d_tid].startOfStack, d_stacksize);
-    pdns_swapcontext(d_kernel, *d_threads[d_tid].context);
+    try {
+      pdns_swapcontext(d_kernel, *d_threads[d_tid].context);
```

Comment:
Don't we need the same construct a few lines below, and in `::sendEvent()` as well?

Reply:
yes

Reply:
It looks like the new commit you pushed only does the `sendEvent()` case, and not the one on line 370.